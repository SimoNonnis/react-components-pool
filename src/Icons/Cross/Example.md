```jsx
<Cross fill="#000"/>
```
