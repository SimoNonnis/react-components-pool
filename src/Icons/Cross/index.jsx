import React from 'react';
import PropTypes from 'prop-types';

function Cross({ width, height, fill, className, classNamePath }) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      width={width}
      height={height}
      className={className}
    >
      <title>Cross Icon</title>

      <polygon
        className={classNamePath}
        fill={fill}
        points="19.04 6.38 17.62 4.96 12 10.59 6.38 4.96 4.96 6.38 10.59 12 4.96 17.62 6.38 19.04 12 13.41 17.62 19.04 19.04 17.62 13.41 12 19.04 6.38"
      />
    </svg>
  );
}

Cross.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  fill: PropTypes.string,
  className: PropTypes.string,
  classNamePath: PropTypes.string,
};

Cross.defaultProps = {
  width: 32,
  height: 32,
  fill: '#fff',
  className: 'chainIcon',
  classNamePath: 'chainIconPath',
};

export default Cross;
