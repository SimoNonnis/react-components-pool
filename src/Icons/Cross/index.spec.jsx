/* eslint-env jest */
import React from 'react';

import Cross from './';

describe('Test <Cross /> icon', () => {
  it('should render Cross icon correctly with default props', () => {
    const wrapper = shallow(<Cross />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Cross icon correctly with props', () => {
    const wrapper = shallow(<Cross fill="#a4a1a1" className="pof" height={30} width={30} />);
    expect(wrapper).toMatchSnapshot();
  });
});
