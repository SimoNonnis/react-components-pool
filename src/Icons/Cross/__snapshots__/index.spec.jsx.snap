// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Cross /> icon should render Cross icon correctly with default props 1`] = `
<svg
  className="chainIcon"
  height={32}
  viewBox="0 0 24 24"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Cross Icon
  </title>
  <polygon
    className="chainIconPath"
    fill="#fff"
    points="19.04 6.38 17.62 4.96 12 10.59 6.38 4.96 4.96 6.38 10.59 12 4.96 17.62 6.38 19.04 12 13.41 17.62 19.04 19.04 17.62 13.41 12 19.04 6.38"
  />
</svg>
`;

exports[`Test <Cross /> icon should render Cross icon correctly with props 1`] = `
<svg
  className="pof"
  height={30}
  viewBox="0 0 24 24"
  width={30}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Cross Icon
  </title>
  <polygon
    className="chainIconPath"
    fill="#a4a1a1"
    points="19.04 6.38 17.62 4.96 12 10.59 6.38 4.96 4.96 6.38 10.59 12 4.96 17.62 6.38 19.04 12 13.41 17.62 19.04 19.04 17.62 13.41 12 19.04 6.38"
  />
</svg>
`;
